lib/food-generator.js

```diff
@@ -7,36 +7,56 @@ function FoodGenerator(width, height) {
 }
 
 FoodGenerator.prototype = {
-  seedFood: function() {
+  seedFood: function(players) {
     var food = [];
     for(var i = 0; i < 50; i++) {
-      var foodItem = new Food(this.randOnCanvas());
+      var foodItem = new Food(this.randOnCanvas(players));
       food.push(foodItem);
     }
     return food;
   },
 
-  seedSpeedBoosts: function() {
+  seedSpeedBoosts: function(players) {
     var speedBoosts = [];
     for(var i = 0; i < 4; i++) {
-      var boost = new SpeedBoost(this.randOnCanvas());
+      var boost = new SpeedBoost(this.randOnCanvas(players));
       speedBoosts.push(boost);
     }
     return speedBoosts;
   },
 
-  randOnCanvas: function() {
-    return { x: Math.floor(Math.random() * (this.canvasWidth - 10) + 5),
-             y: Math.floor(Math.random() * (this.canvasHeight - 10) + 5) };
+  randOnCanvas: function(players) {
+    var newCoords =  { x: Math.floor(Math.random() * (this.canvasWidth - 10) + 5),
+                       y: Math.floor(Math.random() * (this.canvasHeight - 10) + 5) };
+    var isNotInsidePlayer = false;
+    var canWidth = this.canvasWidth;
+    var canHeight = this.canvasHeight;
+
+    function checkPlayer(player, i, players){
+      var xDiff = newCoords.x - player.x;
+      var yDiff = newCoords.y - player.y;
+      var distance = Math.sqrt( xDiff*xDiff + yDiff*yDiff);
+      return distance > (player.mass + 15);
+    }
```

Comment:
Let's accept this, as it works and makes sense. I'm going to add a waffle issue to look into refactoring. Seems like we're duplicating the collision detection logic and giving it to both the player and the food generator. That should live in one spot, and either iterate through players to have them check for collision or create an object in charge of checking the collision between objects.